### ISSUE TITLE
Rule proposal: `require-proxy-set-returns-true`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2178]
### Description

According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy/set#return_value

> If the `set()` method returns `false`, and the assignment happened in strict-mode code, a [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError) will be thrown.

The following code will throws.

```js
(function () {
    "use strict";
    const proxy = new Proxy({}, {
        set(target, property, value) {
        }
    });
    proxy.foo = 'bar'
})()

// Uncaught TypeError: 'set' on proxy: trap returned falsish for property 'foo'
```

This rule enforce the `set` to explicitly return `true`. 
This rule also allow return [`Reflect.set()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/set) call.

### Fail

```js
const proxy = new Proxy({}, {
    set(target, property, value) {
        target[property] = value;
    }
});
````

```js
const proxy = new Proxy({}, {
    set(target, property, value) {
        target[property] = value;
        return 1; // Will work, but should be `true` to be clear.
    }
});
````

### Pass

```js
const proxy = new Proxy({}, {
    set(target, property, value) {
        target[property] = value;
        return true;
    }
});
````

```js
const proxy = new Proxy({}, {
    set(target, property, value) {
        return Reflect.set(...arguments);
    }
});
````

**Comment by fisker** [comment]
Run into the same problem again, turns out [`handler.setPrototypeOf()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy/setPrototypeOf#return_value)  [`handler.deleteProperty()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy/deleteProperty#return_value),  [`handler.preventExtensions()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy/preventExtensions#return_value) need return `true` too.

Didn't test, but [`handler.isExtensible()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy/isExtensible#return_value) also throws in some case.

**Comment by sindresorhus** [comment]
Accepted
